src/Analyzer.JsonRuleEngine/Operators/InequalityOperator.cs

```diff
@@ -0,0 +1,132 @@
+        /// Evaluates <paramref name="tokenToEvaluate"/> to determine if it is greater, greaterOrEquals, less, or lessOrEquals than the specified value.
+        /// </summary>
+        /// <param name="tokenToEvaluate">The JToken to evaluate.</param>
+        /// <returns>A value indicating whether or not the evaluation passed.</returns>
+        public override bool EvaluateExpression(JToken tokenToEvaluate)
+        {
+            if (tokenToEvaluate == null)
+            {
+                // If the specified property in the JSON is not defined then we would assume it could potentially have an undesired value
+                return false;
+            }
+
+            ValidateComparisonTerm(tokenToEvaluate);
+
+            // TODO throw this exception earlier?
+            if ((SpecifiedValue.Type == JTokenType.Date && tokenToEvaluate.Type != JTokenType.Date) ||
+                (tokenToEvaluate.Type == JTokenType.Date && SpecifiedValue.Type != JTokenType.Date))
+            {
+                throw new InvalidOperationException($"Cannot compare {SpecifiedValue.Type} with {tokenToEvaluate.Type} using an InequalityOperator");
+            }
+
+            var normalizedSpecifiedValue = GetNormalizedValue(SpecifiedValue);
+            var normalizedTokenToEvaluate = GetNormalizedValue(tokenToEvaluate);
+
+            var result = Greater ? normalizedSpecifiedValue > normalizedTokenToEvaluate : normalizedSpecifiedValue < normalizedTokenToEvaluate;
+
+            if (OrEquals)
+            {
+                result = result || normalizedSpecifiedValue == normalizedTokenToEvaluate;
+
+            }
+
+            return result;
+        }
+
+        // TODO throw this exception earlier?
+        private void ValidateComparisonTerm(JToken term)
+        {
+            var validTypes = new JTokenType[] { JTokenType.Date, JTokenType.Float, JTokenType.Integer };
+
+            if (!validTypes.Contains(term.Type))
+            {
+                throw new InvalidOperationException($"Cannot compare against a {term.Type} using an InequalityOperator");
+            }
+        }
+
+        private double GetNormalizedValue(JToken token)
+        {
+            double value;
+
+            if (token.Type == JTokenType.Date) // TODO test and document JSON date format?
+            {
+                value = token.Value<DateTime>().ToOADate();
+            }
+            else
+            {
+                value = token.Value<double>();
+            }
+
+            return value;
+        }
+
+        private string GetName() {
+            if (Greater && OrEquals)
+            {
+                return "GreaterOrEquals";
+            }
+            else if (Greater && !OrEquals)
+            {
+                return "Greater";
+            }
+            else if (!Greater && OrEquals)
+            {
+                return "LessOrEquals";
+            }
+            else
+            {
+                return "Less";
+            }
+        }
```

Comment:
This can be simplified a bit and look tidier :)
``` suggestion
        private string GetName() =>
            Greater && OrEquals ? "GreaterOrEquals" :
            Greater && !OrEquals ? "Greater" :
            !Greater && OrEquals ? "LessOrEquals" :
            "Less";
```

Reply:
I didn't commit the suggestion because I rather not define the function if we are doing it this way, but made the change